packages/devtools_flutter/lib/src/config.dart

```diff
@@ -10,28 +11,111 @@ import 'screen.dart';
 
 /// Top-level configuration for the app.
 @immutable
-class Config {
-  /// The routes that the app exposes.
-  Map<String, WidgetBuilder> get routes {
-    final routeToBuilder = <String, WidgetBuilder>{};
-    for (var key in _routeToTabs.keys) {
-      routeToBuilder[key] =
-          (BuildContext context) => DevToolsScaffold(tabs: _routeToTabs[key]);
+class DevToolsApp extends StatefulWidget {
+  @override
+  State<DevToolsApp> createState() => DevToolsAppState();
+}
+
+/// The DevTools app's state.
+///
+/// This manages the route generation, and marshalls URL query parameters into flutter route parameters.
+// TODO(https://github.com/flutter/devtools/issues/1146): Introduce tests that navigate the full app.
+class DevToolsAppState extends State<DevToolsApp> {
+  @override
+  void initState() {
+    super.initState();
+    service.FrameworkCore.init(WidgetsBinding.instance.window.defaultRouteName);
+  }
+
+  /// Generates routes, separating the path from URL query parameters.
+  Route generateRoute(RouteSettings settings) {
+    final uri = Uri.parse(settings.name);
+    final queryParams = uri.queryParameters;
+    final path = uri.path;
+    print('path: $path, params: $queryParams');
```

Comment:
is this just for debugging? If so, please remove before merging.

Reply:
Good catch, done.